packages/file_selector/file_selector_linux/linux/file_selector_plugin.cc

```diff
@@ -192,7 +196,8 @@ static void method_call_cb(FlMethodChannel* channel, FlMethodCall* method_call,
   FlValue* args = fl_method_call_get_args(method_call);
 
   g_autoptr(FlMethodResponse) response = nullptr;
-  if (strcmp(method, kOpenFileMethod) == 0) {
+  if (strcmp(method, kOpenFileMethod) == 0 ||
+      strcmp(method, kGetDirectoriesPathsMethod) == 0) {
```

Comment:
The logic to extract one value from a list of returns should be on the Dart side, as it is for files.